docs/pages/reference/cli.mdx

```diff
@@ -1423,6 +1423,93 @@ DEBU [TSH]       Self re-exec command: tsh [status --format=json]. tsh/aliases.g
 ...
 ```
 
+#### Proxy Templates
+
+With Proxy Templates, `tsh` will dynamically determine the address of the
+Proxy Service to connect to based on the address of the destination host
+in your `tsh ssh` or `tsh proxy ssh` command.
+
+```yaml
+proxy_templates:
+
+# Regular expression that the host server address `%h:%p` is matched against.
+# The "replace rules" below can reference capturing groups from this regular
+# expression (`$1`, `$2`, etc.).
+- template: '^(\w+)\.(\w+):([0-9]+)$' # <nodename>.<clustername>:<port>
+
+  # Optional web proxy address to use for proxy jump (`--jumphost`, `-J`).
+  #
+  # Proxy Jump can be used to reduce latency in regionally distributed trusted
+  # clusters by connecting to a leaf node through the leaf proxy instead of the
+  # root proxy.
+  proxy: "$2.eu.example.com:443"
+
+  # Optional cluster name to connect to (`--cluster`).
+  #
+  # Cluster can be used to connect to leaf nodes from the root proxy without
+  # first logging in to the leaf cluster. This may be useful in cases where
+  # proxy jump is not applicable, such as when the leaf clusters do not have
+  # their own public proxies.
+  cluster: "$2"
+
+  # Optional host server address to connect to (`%h:%p`).
+  #
+  # Port defaults to 3022 if not explicitly provided with `--port`.
+  host: "$1:$3"
+
+# Multiple templates can be provided. They are evaluated in order and the first
+# one matching will take effect.
+- template: ... 
+```
+
+`tsh -J {{proxy}} ssh` and `tsh -J {{proxy}} proxy ssh` will attempt to match the 
+host server address `%h:%p` with the configured templates. For each replace rule set, 
+the corresponding cli value will be set.
+
+If leaf certificates are required to connect to the node, `tsh` will automatically
+retrieve leaf certificates from the root cluster.
+
+```bash
+$ tsh ssh -J {{proxy}} node1.leaf1
+# becomes
+$ tsh ssh -J leaf1.eu.example.com:443 --cluster leaf1 node1
+```
+
+If there is no template matched, an error is returned.
+
+```bash
+$ tsh ssh -J {{proxy}} node1.none.example.com
+ERROR: proxy jump contains {{proxy}} variable but did not match any of the templates in tsh config
+```
+
+If the proxy variable `-J {{proxy}}` is not explicitly provided, then `tsh` will 
+still attempt to match a template, but will not fail when there is no match.
+Additionally, it will not replace the `proxy` or `cluster` values when explicitly
+set by the client.
+
+```bash
+$ tsh ssh -J leaf2.us.example.com:443 node1.leaf2
+# becomes
+$ tsh ssh -J leaf2.us.example.com:443 --cluster leaf2 node1
+```
+
+Proxy Templates can also be used with OpenSSH by setting the `ProxyCommand` 
+in `~/.ssh/config` to use `tsh proxy ssh`.
+
+```txt
+Host *.example.com
+    Port 3022
+    ProxyCommand tsh proxy ssh -J {{proxy}} %r@%h:%p
+```
+
+As a result, you can use `tsh ssh` and `ssh` interchangeably.
+
+```bash
```

Comment:
The renderer expects `code` to apply command formatting:

````suggestion
```code
$ tsh ssh -J {{proxy}} node1.leaf1
# becomes
$ tsh ssh -J leaf1.eu.example.com:443 --cluster leaf1 node1
```

If there is no template matched, an error is returned.

```code
$ tsh ssh -J {{proxy}} node1.none.example.com
ERROR: proxy jump contains {{proxy}} variable but did not match any of the templates in tsh config
```

If the proxy variable `-J {{proxy}}` is not explicitly provided, then `tsh` will 
still attempt to match a template, but will not fail when there is no match.
Additionally, it will not replace the `proxy` or `cluster` values when explicitly
set by the client.

```code
$ tsh ssh -J leaf2.us.example.com:443 node1.leaf2
# becomes
$ tsh ssh -J leaf2.us.example.com:443 --cluster leaf2 node1
```

Proxy Templates can also be used with OpenSSH by setting the `ProxyCommand` 
in `~/.ssh/config` to use `tsh proxy ssh`.

```txt
Host *.example.com
    Port 3022
    ProxyCommand tsh proxy ssh -J {{proxy}} %r@%h:%p
```

As a result, you can use `tsh ssh` and `ssh` interchangeably.

```code
````